### ISSUE TITLE
Model class methods need to be overrided caused by Pytorch Lightning

### ISSUE THREAD

**Issue by Tony-Tseng** [akshaydudhane16/BIPNet#2]
Thanks for your work. 
I'm trying to reproduce the training result for super-resolution track 1.
However, after installing the environment by install.yml, I cannot run the training code by 
`python BIPNet_Track_1_training.py`.

The error appears as follows.
```
"No `training_step()` method defined. Lightning `Trainer` expects as minimum a"
pytorch_lightning.utilities.exceptions.MisconfigurationException: No `training_step()` method defined. Lightning `Trainer` expects as minimum a `training_step()`, `train_dataloader()` and `configure_optimizers()` to be defined.
```

I'm fixing it by adding training_step, and configure_optimizers functions in BIPNet and setting Adam optimizer with lr 1e-4 and Cosine annealing scheduler as the paper states. Is it the correct way to reproduce the result?

**Comment by akshaydudhane16** [comment]
Issue is fixed by updating Network.py file.

**Comment by Tony-Tseng** [comment]
The code works! Thanks for the update.
